### PR TITLE
GitHub Issue #140: Refactor usage of levels in log methods

### PR DESCRIPTION
**Pull request by ArturMoczulski**
This removes most of the usage of static calls to Level class and clarifies the confusion between using `Psr\Log\Level` and `Rollbar\Payload\Level`.

https://github.com/rollbar/rollbar-php/issues/140

### CODE REVIEW

File: src/Payload/Level.php [hunk 1]
```diff
@@ -2,6 +2,17 @@
 
 class Level implements \JsonSerializable
 {
+    const IGNORE = 'ignore';
```
Comment (coryvirok) [inline src/Payload/Level.php]: What are these two "ignore(d)" levels for? They don't map to any level we have on the backend. Same for "alert" and "emergency".

**Comment by coryvirok** [comment]
Much cleaner!

File: src/Payload/Level.php (hunk 1 above)
Comment (ArturMoczulski) [inline src/Payload/Level.php]: `IGNORE` and `IGNORED` are synonymous and they can be used to completely silence the logger if used in `minimumLevel` config option. I think it's mostly a leftover from 0.18.2, I don't really see the point and it's also missing from README.md. I highly doubt anybody is using them so I'm happy to remove them.

`ALERT` and `EMERGENCY` are PSR-3 levels and are mapped to Rollbar levels here: https://github.com/rollbar/rollbar-php/blob/master/src/Payload/Level.php#L16-L33

Comment (coryvirok) [inline src/Payload/Level.php]: Aha! Thanks for the info. WOuld you mind adding a comment to this code and point to where they are mapped to the Rollbar levels?

**Comment by ArturMoczulski** [comment]
This is ready for review @coryvirok @rokob

File: tests/LevelTest.php [hunk 2]
```diff
@@ -7,17 +7,22 @@ class LevelTest extends \PHPUnit_Framework_TestCase
 {
     public function testLevel()
     {
-        $l = Level::TEST();
-        $this->assertNull($l);
+        try {
+            $level = Level::TEST();
+            $this->fail();
+        } catch (\Exception $exception) {
+            $this->assertTrue(true);
```
Comment (olafnorge) [inline tests/LevelTest.php]: What is the sense behind this block? I don't what is tested here.

Comment (rokob) [inline tests/LevelTest.php]: This appears to be testing that `Level::TEST()` throws an exception. The alternative would be to use `expectException()`

**Comment by rokob** [comment]
This looks good to me, we will need to have a bump in version beyond the bugfix level because this changes the API

**Comment by rokob** [comment]
Wait I am wrong about that, this uses the `__callStatic` metaprogramming magic to patch back those calls and redirect them to the constants so this is still api compatible right?

**Comment by ArturMoczulski** [comment]
@rokob Yes, this is backward compatible. The static methods still work but I marked that as deprecated in favor of `Level::` constants. If all is clear, let's get this merged.

File: tests/LevelTest.php (hunk 2 above)
Comment (ArturMoczulski) [inline tests/LevelTest.php]: @olafnorge this tests that an exception is thrown if a static method is called on `Level` that does not match any of the logging levels. Mostly this doesn't matter, as static methods are now deprecated.

Comment (olafnorge) [inline tests/LevelTest.php]: Ahh, thanks for clarifying.

**Pull request merged**